lib/features/digital_guide_view/data/repository/digital_guide_repository.dart

```diff
@@ -0,0 +1,54 @@
+import "package:dio/dio.dart";
+import "package:flutter/foundation.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+import "../../../../api_base_rest/client/dio_client.dart";
+import "../../../../config/env.dart";
+import "../models/digital_guide_response.dart";
+
+part "digital_guide_repository.g.dart";
+
+@riverpod
+
+@riverpod
+Future<DigitalGuideResponse> getDigitalGuideData(Ref ref, int id) async {
+
+  debugPrint("start fetching digital guide data");
+
+  try {
+
+    final digitalGuideUrl = "${Env.digitalGuideUrl}/buildings/$id";
+    debugPrint("Url: $digitalGuideUrl");
+    debugPrint("Authorization token: ${Env.digitalGuideAuthorizationToken}");
+    final dio = ref.read(restClientProvider);
+    dio.options.headers["Authorization"] = "Token ${Env.digitalGuideAuthorizationToken}";
```

Comment:
we're also planning on introducing some caching, but again, you don't need to worry about this in this task

Reply:
Out of curiosity, doesn't Riverpod deal with caching?

Reply:
@24bartixx  it deals with caching but during single run of an app. We want stronger cache, called persistent cache that will also keep the data when you exit and reeenter the app next day. We basically just drop the data to local DB and if we have quite recent data in the db, we then query this, instead of a network call. Obviosuly cache invalidation then is needed to flush some out of date data, but the cache invalidation is actually one of the two hardest things in the Computer Science (this comes from famous quote: https://skeptics.stackexchange.com/questions/19836/has-phil-karlton-ever-said-there-are-only-two-hard-things-in-computer-science)